lib/vty.c

```diff
@@ -2077,7 +2069,6 @@ static int vtysh_read(struct thread *thread)
 
 	sock = THREAD_FD(thread);
 	vty = THREAD_ARG(thread);
-	vty->t_read = NULL;
```

Comment:
If you remove `t_read` there is a situation later in this function that returns without setting this to `NULL`. This will cause later schedules to fail because this variable will be set.

This is the situation:

```c
				if (!vty->t_write && (vtysh_flush(vty) < 0))
					/* Try to flush results; exit if a write
					 * error occurs. */
return 0;
```

Reply:
@rzalamena -> you do realize that it is already set to NULL?